test/Microsoft.Sbom.Targets.Tests/AbstractGenerateSbomTaskTests.cs

```diff
@@ -0,0 +1,264 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Sbom.Api.Utils;
+using Microsoft.Sbom.Contracts;
+using Microsoft.Sbom.Extensions.DependencyInjection;
+using Microsoft.Sbom.Targets;
+using Microsoft.Sbom.Targets.Tests.Utility;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Newtonsoft.Json;
+
+namespace Microsoft.Sbom.Targets.Tests;
+
+/// <summary>
+/// Base class for testing SBOM generation through the GenerateSbomTask.
+/// </summary>
+[TestClass]
+public abstract class AbstractGenerateSbomTaskTests
+{
+    internal SbomSpecification SbomSpecification;
+
+    private static readonly string CurrentDirectory = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
```

Comment:
nit: Should these variables be public/internal so child classes can utilize them?

Reply:
Thanks. Changed at 25a386d0a05ec60d5b2d03b952c203dc2ade8bd2